src/store.ts

```diff
@@ -0,0 +1,15 @@
+import { configureStore } from '@reduxjs/toolkit';
+import createDocReducer from './containers/CreateDoc/createDocSlice';
+import { useSelector, TypedUseSelectorHook } from 'react-redux';
+
+type RootState = ReturnType<typeof store.getState>;
```

Comment:
Can you explain what'a happening over here ? 

What is a ReturnType?
Why does it need an argument  ?
Why that argument is a `typeof` return of a `store.getState` ?

Reply:
It's a typescript utility, which constructs a type consisting of the return type of function:
https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypet

To set properly the root reducer state, since we pass the slice reducers directly to configureStore(), we never created the root reducer ourselves, so we need to infer it as type RootState = ReturnType<typeof store.getState> instead. 

If we were using combineReducer, for example:

const rootReducer = combineReducers({
  createDoc: createDocReduce,
})

We would have defined the type this way: type RootState = ReturnType<typeof rootReducer>

More info: https://redux.js.org/recipes/usage-with-typescript#type-checking-reducers